### ISSUE TITLE
ckeditor-duplicated-modules error after upgrading Node & CKEditor packages

### ISSUE THREAD

**Issue by Grkmk** [ckeditor/ckeditor5#15720]
After upgrading Node, React and the CKEditor packages, our editor implementation (from the source) has started to give the duplicated-modules error.

I went over the error documentation to see if there was something we misconfigured for the newer versions. It helped me fix some webpack issues we were initially having, but I believe I've exhausted all the cases & solutions listed on the error's page (i.e. mismatched plugin versions, not building from the source & adding plugins).

I'm guessing it's related to our Webpack setup, but I'm unable to locate any hints in the documentation regarding the error. I'm not sure if it's indeed a bug, but if it is, I would be more than happy to open a PR for it -- with your help of course, I have no familiarity with the CKEditor codebase.

I've created a [repository](https://github.com/Grkmk/ckeditor-issue) with the minimum reproducible code. I would greatly appreciate it if you could take a look at it.

* Browser: chrome 119.0.6045.199 (Official Build) (arm64)
* OS: macOS 14.2
* Node: 20.10
* React: 18.2
* First affected CKEditor version: 40.2
* Installed CKEditor plugins: 40.2

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by olia-bn** [comment]
Same here

**Comment by Witoso** [comment]
Two things that helped me to get this working, I achieved them by trial-and-error, as I didn't have an opportunity to analyze the full webpack config:

1.  use `npm install --legacy-peer-deps` , as we released `v41` already, `npm` has a big trouble with resolving correctly the deps and duplicates the packages versions.
2.  move CKEditor packages to `dependencies` frome `devDependencies` , more of a hunch, but sometimes bundlers do weird things with devDependencies.

I didn't push the fix branch as I don't have permissions.

Cheers!
